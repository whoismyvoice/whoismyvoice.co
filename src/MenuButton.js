import React, { Component, } from 'react';
import PropTypes from 'prop-types';
import { Link, } from 'react-router-dom';
import { connect, } from 'react-redux';
import cx from 'classnames';

import {
  toggleMenu,
} from './actions';
// Styles
import './styles/MenuButton.css';

export class MenuButton extends Component {
  static defaultProps = {
    isMenuOpen: false,
    onMenuButtonClick: () => {},
  }

  static propTypes = {
    isMenuOpen: PropTypes.bool,
    onMenuButtonClick: PropTypes.func,
  }

  handleClick(event) {

  }

  handleRestart(event) {

  }

  render() {
    const {
      didSearch,
      isMenuOpen,
      onMenuButtonClick,
    } = this.props;
    const menuClasses = cx(
      [ 'menu-overlay', ],
      {
        'menu-overlay--show': isMenuOpen && didSearch,
        'menu-overlay--white': isMenuOpen && !didSearch,
      },
    );
    const buttonClasses = cx(
      [ 'menu-buttons', 'animated', ],
      {
        'bounceInRight': isMenuOpen,
        'bounceOutRight': !isMenuOpen,
        'menu-buttons--white': !didSearch,
      },
    );
    const menuButtonClasses = cx(
      [ 'menu-button', 'animated', ],
      {
        'menu-button--clicked': isMenuOpen,
      },
    );

    return (
    <React.Fragment>
      <div className={menuClasses}>
        <div className="menu-container">
          <Link to="/sources" className={buttonClasses}>
            Data sources
          </Link>
          <Link to="/about" className={buttonClasses}>
            About this project
          </Link>
          <Link to="/" className={buttonClasses}>
            Start Again
          </Link>
        </div>
      </div>
      <div
        className={menuButtonClasses}
        onClick={onMenuButtonClick}
      >
      </div>
    </React.Fragment>
    );
  }
}

function mapStateToProps(state) {
  const { address, view, } = state;
  return {
    didSearch: address.value !== undefined,
    ...view,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    onMenuButtonClick: (event) => {
      event.preventDefault();
      dispatch(toggleMenu());
    },
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(MenuButton);
