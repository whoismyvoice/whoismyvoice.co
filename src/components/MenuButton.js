import React from 'react';
import { Link } from 'react-router';
import cx from 'classnames';
import SenateStore from '../stores/SenateStore';
import SenateActions from '../actions/SenateActions';

// Components
import BaseComponent from './BaseComponent';

// Styles
import style from '../styles/MenuButton.scss';

class MenuButton extends BaseComponent {
  constructor() {
    super();
    this._bind('_toggleOverlay', '_handleChange', '_destroyFullpage', '_handleRestart');
    this.state = SenateStore.getMember();
  }

  _handleRestart() {
    this.setState({didClick: !this.state.didClick});
    SenateActions.flush();
    this._destroyFullpage();
  }
  _destroyFullpage() {
    if ($.fn.fullpage.destroy !== undefined) {
      $.fn.fullpage.destroy();
    }
  }
  componentDidMount() {
    SenateStore.addChangeListener(this._handleChange);
  }
  componentWillUnmount() {
    SenateStore.removeChangeListener(this._handleChange);
  }
  _handleChange() {
    this.setState(SenateStore.getMember());
  }

  _toggleOverlay() {
    this.setState({didClick: !this.state.didClick});
  }

  render() {
    const menuClasses = cx(
      ['menu-overlay'],
      {'menu-overlay--show': this.state.didClick && this.state.did_search},
      {'menu-overlay--white': this.state.didClick && !this.state.did_search}
    );

    const buttonClasses = cx(
      ['menu-buttons', 'animated'],
      {'bounceInRight': this.state.didClick},
      {'bounceOutRight': !this.state.didClick},
      {'menu-buttons--white': !this.state.did_search}
    );

    const menuButtonClasses = cx(
      ['menu-button', 'animated'],
      {'menu-button--clicked': this.state.didClick}
    );

    return <span>
      <div className={menuClasses}>
        <div className="menu-container">
          <Link to="/sources">
            <div className={buttonClasses} onClick={this._toggleOverlay}>
              Data sources
            </div>
          </Link>
          <Link to="/about">
            <div className={buttonClasses} onClick={this._toggleOverlay}>
              About this project
            </div>
          </Link>
          <Link to="/">
            <div className={buttonClasses} onClick={this._handleRestart}>
              Start Again
            </div>
          </Link>
        </div>
      </div>
      <div
        className={menuButtonClasses}
        onClick={this._toggleOverlay}
      >
      </div>
    </span>;
  }
}

export default MenuButton;
