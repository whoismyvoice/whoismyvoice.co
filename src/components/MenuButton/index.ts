import ConnectedMenuButton from './MenuButton';

export default ConnectedMenuButton;
