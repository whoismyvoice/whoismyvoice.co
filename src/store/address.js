import {
  RECEIVE_ADDRESS,
  RECEIVE_GPS,
  RECEIVE_ZIP_CODE,
  RESET_CURRENT,
} from '../actions/types';

const initialState = {
  value: undefined,
  street: undefined,
  zipCode: undefined,
  latitude: undefined,
  longitude: undefined,
};

function handle(state = initialState, action) {
  const { type } = action;
  switch (type) {
    case RECEIVE_ADDRESS:
      return {
        ...state,
        street: action.address.replace(new RegExp(`(, )?${state.zipCode}`), ''),
        value: action.address,
      };
    case RECEIVE_GPS:
      return {
        ...state,
        latitude: action.latitude,
        longitude: action.longitude,
      };
    case RECEIVE_ZIP_CODE:
      return {
        ...state,
        zipCode: action.zipCode,
      };
    case RESET_CURRENT:
      return initialState;
    default:
      return state;
  }
}

export default handle;
