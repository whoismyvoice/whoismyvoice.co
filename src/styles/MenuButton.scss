@import '_variables';
@import '_transitions';

.menu-button {
  z-index: 10;
  position: fixed;
  display: block;
  -moz-box-shadow: 3px 3px 5px rgba(68, 68, 68, 0.2);
  -webkit-box-shadow: 3px 3px 5px rgba(68, 68, 68, 0.2);
  box-shadow: 3px 3px 5px rgba(68, 68, 68, 0.2);
  width: 35px;
  height: 35px;
  right: 35px;
  bottom: 35px;
  background-color: $white;
  border-radius: 50%;
  text-align: center;
  line-height: 2.2;
  cursor: pointer;

  &:before,
  &:after {
    @include transition(0.2s);
    content: '?';
    font-weight: bold;
    font-size: 16px;
    text-align: center;
    width: 20px;
    height: 20px;
    color: $black;
    font-family: 'Open Sans';
  }

  &:after {
    content: 'x';
    font-family: 'Dosis';
    font-size: 22px;
    line-height: 1.2;
    position: absolute;
    left: 22%;
    top: 2%;
    opacity: 0;
  }

  &:hover {
    &:before,
    &:after {
      color: $dark-orange;
    }
  }

  @media screen and (max-width: $iphone6) {
    right: 20px;
    bottom: 20px;
  }
}

.menu-button--clicked {
  &:after {
    @include transition(0.2s);
    opacity: 1;
  }

  &:before {
    opacity: 0;
  }
}

.menu-overlay {
  @include transition(0.5s);
  visibility: hidden;
  position: fixed;
  height: 100vh;
  width: 100vw;
  display: block;
  z-index: 4;
  background-color: rgba(255, 149, 73, 0);

  a {
    color: $black;
  }
}

.menu-overlay--show {
  @include transition(0.5s);
  visibility: visible;
  background-color: rgba(255, 149, 73, 0.9);
}

.menu-overlay--white {
  @include transition(0.5s);
  visibility: visible;
  background-color: rgba(255, 255, 255, 0.2);

  @media screen and (max-width: $iphone6) {
    background-color: rgba(255, 255, 255, 0.5);
  }
}

.menu-container {
  position: fixed;
  right: 35px;
  bottom: 115px;
  width: 185px;
  z-index: 5;
  height: 85px;
  display: inline-block;

  @media screen and (max-width: $iphone6) {
    right: 20px;
    bottom: 100px;
  }
}

.menu-buttons {
  @include transition(0.5s);
  padding: 5px 10px 5px 10px;
  background-color: $white;
  border-radius: 4px;
  display: block;
  font-family: 'Open Sans';
  font-weight: 600;
  text-align: right;
  font-size: 12px;
  float: right;
  position: relative;
  text-transform: uppercase;
  margin-bottom: 15px;

  &.animated {
    -webkit-animation-duration: 1.5s;
    animation-duration: 1.5s;
  }

  &:hover {
    @include transition(0.2s);
    color: $dark-orange;
  }
}

.menu-buttons--white {
  -moz-box-shadow: 2px 2px 2px rgba(68, 68, 68, 0.2);
  -webkit-box-shadow: 2px 2px 2px rgba(68, 68, 68, 0.2);
  box-shadow: 2px 2px 2px rgba(68, 68, 68, 0.2);
}
