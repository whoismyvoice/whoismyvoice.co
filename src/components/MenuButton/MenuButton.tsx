import React from 'react';
import { Link } from 'react-router-dom';
import { connect, ConnectedProps } from 'react-redux';
import cx from 'classnames';

import { toggleMenu, reset } from '../../actions';
import { Dispatch } from '../../actions/types';
import { State } from '../../store';
import { ViewState } from '../../store/view';
// Styles
import '../../styles/MenuButton.scss';

function mapStateToProps(state: State): StateProps & ViewState {
  const { address, view } = state;
  return {
    didSearch: address.value !== undefined,
    ...view,
  };
}

function mapDispatchToProps(dispatch: Dispatch): DispatchProps {
  return {
    onMenuButtonClick: (event: React.MouseEvent<HTMLDivElement>) => {
      event.preventDefault();
      dispatch(toggleMenu());
    },
    onMenuLinkClick: (event: React.MouseEvent<HTMLAnchorElement>) => {
      dispatch(toggleMenu());
    },
    onRestartClick: (event: React.MouseEvent<HTMLAnchorElement>) => {
      dispatch(reset());
      dispatch(toggleMenu());
    },
  };
}

const connector = connect(mapStateToProps, mapDispatchToProps);
type MappedProps = ConnectedProps<typeof connector>;

interface StateProps {
  didSearch?: boolean;
}

interface DispatchProps {
  onMenuButtonClick?: (event: React.MouseEvent<HTMLDivElement>) => void;
  onMenuLinkClick?: (event: React.MouseEvent<HTMLAnchorElement>) => void;
  onRestartClick?: (event: React.MouseEvent<HTMLAnchorElement>) => void;
}

export class MenuButton extends React.Component<MappedProps> {
  static defaultProps = {
    didSearch: false,
    isMenuOpen: false,
    onMenuButtonClick: () => undefined,
    onMenuLinkClick: () => undefined,
    onRestartClick: () => undefined,
  };

  render() {
    const {
      didSearch,
      isMenuOpen,
      onMenuButtonClick,
      onMenuLinkClick,
      onRestartClick,
    } = this.props;
    const menuClasses = cx(['menu-overlay'], {
      'menu-overlay--show': isMenuOpen && didSearch,
      'menu-overlay--white': isMenuOpen && !didSearch,
    });
    const buttonClasses = cx(['menu-buttons', 'animated'], {
      bounceInRight: isMenuOpen,
      bounceOutRight: !isMenuOpen,
      'menu-buttons--white': !didSearch,
    });
    const menuButtonClasses = cx(['menu-button', 'animated'], {
      'menu-button--clicked': isMenuOpen,
    });

    return (
      <React.Fragment>
        <div className={menuClasses}>
          <div className="menu-container">
            <Link
              to="/sources"
              className={buttonClasses}
              onClick={onMenuLinkClick}
            >
              Data sources
            </Link>
            <Link
              to="/about"
              className={buttonClasses}
              onClick={onMenuLinkClick}
            >
              About this project
            </Link>
            <Link to="/" className={buttonClasses} onClick={onRestartClick}>
              Start Again
            </Link>
          </div>
        </div>
        <div className={menuButtonClasses} onClick={onMenuButtonClick} />
      </React.Fragment>
    );
  }
}

export default connector(MenuButton);
