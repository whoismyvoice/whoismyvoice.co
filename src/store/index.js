import thunkMiddleware from 'redux-thunk';
import { createLogger, } from 'redux-logger';
import { applyMiddleware, combineReducers, createStore, } from 'redux';

import address from './address';
import contributions from './contributions';
import officials from './officials';
import view from './view';

const loggerMiddleware = createLogger();

export default createStore(
  combineReducers({
    address,
    contributions,
    officials,
    view,
  }),
  {},
  applyMiddleware(
    thunkMiddleware,
    loggerMiddleware,
  )
);
