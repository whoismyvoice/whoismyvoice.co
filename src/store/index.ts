import thunkMiddleware from 'redux-thunk';
import loggerMiddleware from 'redux-logger';
import { applyMiddleware, combineReducers, createStore } from 'redux';

import address, { AddressState } from './address';
import contributions, { ContributionsState } from './contributions';
import officials, { OfficialsState } from './officials';
import view, { ViewState } from './view';

export interface State {
  address: AddressState;
  contributions: ContributionsState;
  officials: OfficialsState;
  view: ViewState;
};

export default createStore(
  combineReducers({
    address,
    contributions,
    officials,
    view,
  }),
  {},
  applyMiddleware(thunkMiddleware, loggerMiddleware)
);
