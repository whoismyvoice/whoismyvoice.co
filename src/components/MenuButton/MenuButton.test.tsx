import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import MenuButton from './MenuButton';
import store from '../../store';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <Router>
        <MenuButton />
      </Router>
    </Provider>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <MenuButton />
        </Router>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
