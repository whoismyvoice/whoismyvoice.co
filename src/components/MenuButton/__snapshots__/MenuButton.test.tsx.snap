// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="menu-overlay"
  >
    <div
      className="menu-container"
    >
      <a
        className="menu-buttons animated bounceOutRight menu-buttons--white"
        href="/sources"
        onClick={[Function]}
      >
        Data sources
      </a>
      <a
        className="menu-buttons animated bounceOutRight menu-buttons--white"
        href="/about"
        onClick={[Function]}
      >
        About this project
      </a>
      <a
        className="menu-buttons animated bounceOutRight menu-buttons--white"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </div>
  </div>,
  <div
    className="menu-button animated"
    onClick={[Function]}
  />,
]
`;
